apps/api-extractor/src/analyzer/StringChecks.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+/**
+ * Helpers for validating various text string formats.
+ */
+export class StringChecks {
+  // Note: In addition to letters, numbers, underscores, and dollar signs, modern ECMAScript
+  // also allows Unicode categories such as letters, combining marks, digits, and connector punctuation.
+  // These are mostly supported in all environments except IE11, so if someone wants it, we would accept
+  // a PR to allow them (although the test surface might be somewhat large).
+  private static readonly _identifierBadCharRegExp: RegExp = /[^a-z0-9_$]/i;
```

Comment:
This can be simplified to:

```suggestion
  private static readonly _identifierBadCharRegExp: RegExp = /[^\w$]/i;
```

Reply:
Heh, above you asked me to write for an audience that doesn't know what `\w` means.  Here you're asking me to compress the expression.

I'm thinking the compromise is to replace `\w` with `a-z0-9_` as is done here.

Reply:
Let's keep it like this.  Above you didn't seem to know what `\w` did.